mscore/palettebox.h

```diff
@@ -32,10 +34,13 @@ class PaletteBox : public QDockWidget {
       QVBoxLayout* vbox;
       Palette* newPalette(const QString& name, int slot);
       QComboBox* workspaceList;
-      QLineEdit* searchBox;
+
+      //QMap<QString, Shortcut*> localShortcuts;
+
       const int paletteStretch = 1000;
       QAction* singlePaletteAction;
       QToolButton* addWorkspaceButton;
+      bool keyboardNavigation = false;
```

Comment:
This is basically the sense in which we now have "state" for the palette.  It's pretty lightweight and hopefully not a problem.  But since it was something that came up earlier (years ago) as a potential issue, it's worth pointing out here.

Basically, the code ties the keyboard navigation state to the search facility.  We enter this state after a search and then pressing up or down.  It gets cleared if you then apply an element (by hitting Enter).  Not sure if there are other places we should explicitly clear it.  I would think maybe if you clear the search.

Right now, the behavior after clearing the search is a little unclear to me.  If I search for "treble" and navigate through the matching clefs, then clear the search, then open the clefs palette normally, the clef I had navigated to still shows as selected, but doesn't *act* as if it is.  So hitting Enter won't apply it, and hitting Up or Down won't navigate.  So are we still in keyboardNavigation mode or not?  If not, I think we shouldn't still be highlighting the last-selected clef.